src/encoded/audit/experiment.py

```diff
@@ -3920,6 +3920,59 @@ def audit_experiment_nih_institutional_certification(value, system, excluded_typ
         yield AuditFailure('missing nih_institutional_certification', detail, level='ERROR')
 
 
+def audit_experiment_eclip_queried_RNP_size_range(value, system, excluded_types):
+    '''
+    Check if libraries of eCLIP experiment and its control have matching queried_RNP_size_range.
+    '''
+    if value.get('assay_term_name') != 'eCLIP':
+        return
+
+    experiment_size_range = set()
+    control_size_range = set()
```

Comment:
Instead of trying to capture potential differences in ranges of different replicates - you can throw an audit that experiment has mixed size ranges - and "return" after that. Then you can be sure you have only one range you are dealing with